.github/actions/install-poetry/action.yml

```diff
@@ -17,16 +17,16 @@ runs:
     - name: Set up Python
       uses: actions/setup-python@v5
       with:
-        python-version: '3.12'
+        python-version: ${{ matrix.python-version }}
 
     - name: Load cached Poetry installation
       id: cached-poetry
       uses: actions/cache@v4
       with:
         path: |
-          ~/.local
-          ~/.config
-        key: poetry-0
+          ~/.local-${{ matrix.python-version }}
+          ~/.config-${{ matrix.python-version }}
```

Comment:
I'm not sure that this is correct. According to Poetry docs: https://python-poetry.org/docs/configuration/#config-directory, the data and config directories are always under `~/.local/` and `~/.config/`, no matter what the Python version is.